### ISSUE TITLE
More detailed information about limits violated

### ISSUE THREAD

**Issue by udoprog** [spotify/heroic#161]
This suggestion is to change `ResultLimit` into an interface, which has different subtypes depending on the type of limit that was violated.

The `.limits` field would then change from:
```json
{"limits": ["SERIES"]}
```

Into:

```json
{
  "limits": {
    "type": "series",
    "seriesLimit": 20000
  }
}
```

This enhancement would allow the result set to communicate what the limit was, at the time of the query. This information can then be presented to the user, like:

```
Query required too many time series to be processed. The current limit is: 20000.
```

**Comment by parmus** [comment]
Sounds good to me :+1: This would allow for a clearer user experience when hitting these limits.

**Comment by udoprog** [comment]
Closed in favor of #196
